pkg/test/resources/samples/serviceinsights.go

```diff
@@ -0,0 +1,21 @@
+package samples
+
+import (
+	mesh_proto "github.com/kumahq/kuma/api/mesh/v1alpha1"
+	"github.com/kumahq/kuma/pkg/core/resources/apis/mesh"
+	"github.com/kumahq/kuma/pkg/test/resources/builders"
+)
+
+func ServiceInsight(mesh string) *mesh.ServiceInsightResource {
```

Comment:
let's not add arguments like this to samples. You add one argument, then in a couple of months there will be 5 arguments there. Then you have tests that look like this `ServiceInsight("default", 2, true, false, "something)` and adding any new argument means that you need to go over all usages and fix it.
If you want to parameterize anything then you should use builder functions.

do
```
func SampleServiceInsightBuilder() *ServiceInsightBuilder {
	return ServiceInsightBuilderBuilder().
		AddService("test-service", &mesh_proto.ServiceInsight_Service{
			ServiceType: mesh_proto.ServiceInsight_Service_internal,
			Status:      mesh_proto.ServiceInsight_Service_online,
		})
}
```
and call
```
SampleServiceInsightBuilder().WithMesh(mesh).Build()
```

Reply:
but kudos that you actually added builders and samples to this PR. Thanks for that

Reply:
I was meant to clean up this example, thanks for pointing it out.